flang/lib/Lower/Bridge.cpp

```diff
@@ -1576,18 +1605,26 @@ class FirConverter : public Fortran::lower::AbstractConverter {
         if (!isCharSelector)
           return builder->create<mlir::arith::CmpIOp>(loc, pred, selector, rhs);
         fir::factory::CharacterExprHelper charHelper{*builder, loc};
-        auto [lhsAddr, lhsLen] = charHelper.createUnboxChar(selector);
-        auto [rhsAddr, rhsLen] = charHelper.createUnboxChar(rhs);
+        std::pair<mlir::Value, mlir::Value> lhsVal =
+            charHelper.createUnboxChar(selector);
+        mlir::Value &lhsAddr = lhsVal.first;
+        mlir::Value &lhsLen = lhsVal.second;
+        std::pair<mlir::Value, mlir::Value> rhsVal =
+            charHelper.createUnboxChar(rhs);
+        mlir::Value &rhsAddr = rhsVal.first;
+        mlir::Value &rhsLen = rhsVal.second;
```

Comment:
This one clearer makes the code much harder to read. I wonder why that's necessary. <sigh>